wtforms/fields/core.py

```diff
@@ -472,6 +472,12 @@ class SelectField(SelectFieldBase):
     def __init__(self, label=None, validators=None, coerce=text_type, choices=None, validate_choice=True, **kwargs):
         super(SelectField, self).__init__(label, validators, **kwargs)
         self.coerce = coerce
+        if callable(choices):
```

Comment:
Allowing a callable for `choices` is a separate feature request, please make a separate PR.